mypy/main.py

```diff
@@ -102,120 +104,152 @@ def process_options(args: List[str]) -> Tuple[List[BuildSource], Options]:
-            args = args[1:]
-        elif args[0] in ('--silent-imports', '--silent', '-s'):
-            options.build_flags.append(build.SILENT_IMPORTS)
-            args = args[1:]
-        elif args[0] == '--pdb':
-            options.pdb = True
-            args = args[1:]
-        elif args[0] == '--implicit-any':
-            options.implicit_any = True
-            args = args[1:]
-        elif args[0] == '--fast-parser':
-            options.build_flags.append(build.FAST_PARSER)
-            args = args[1:]
-        elif args[0] == '--disallow-untyped-calls':
-            options.build_flags.append(build.DISALLOW_UNTYPED_CALLS)
-            args = args[1:]
-        elif args[0] == '--disallow-untyped-defs':
-            options.build_flags.append(build.DISALLOW_UNTYPED_DEFS)
-            args = args[1:]
-        elif args[0] in ('--version', '-V'):
-            ver = True
-            args = args[1:]
-        else:
-            usage('Unknown option: {}'.format(args[0]))
+    options.dirty_stubs = args.dirty_stubs
+    options.python_path = args.use_python_path
+    options.pdb = args.pdb
+    options.implicit_any = args.implicit_any
+    options.custom_typing_module = args.custom_typing
 
-    if help:
-        usage()
+    # Set build flags.
+    if args.py2:
+        options.pyversion = defaults.PYTHON2_VERSION
 
-    if ver:
-        version()
+    if args.python_version is not None:
+        options.pyversion = args.python_version
 
-    if not args:
-        usage('Missing target file or module')
+    if args.verbose:
+        options.build_flags.append(build.VERBOSE)
 
-    if options.python_path and options.pyversion[0] == 2:
-        usage('Python version 2 (or --py2) specified, '
-              'but --use-python-path will search in sys.path of Python 3')
+    if args.stats:
+        options.build_flags.append(build.DUMP_TYPE_STATS)
 
-    if build.FAST_PARSER in options.build_flags and options.pyversion[0] == 2:
-        usage('The experimental fast parser is only compatible with Python 3, '
-              'but Python 2 specified.')
+    if args.inferstats:
+        options.build_flags.append(build.DUMP_INFER_STATS)
 
-    targets = []
-    for arg in args:
-        if arg.endswith(PY_EXTENSIONS):
-            targets.append(BuildSource(arg, crawl_up(arg)[1], None))
-        elif os.path.isdir(arg):
-            targets.extend(expand_dir(arg))
-        else:
-            targets.append(BuildSource(arg, None, None))
-    return targets, options
+    if args.silent_imports:
+        options.build_flags.append(build.SILENT_IMPORTS)
+
+    if args.disallow_untyped_calls:
+        options.build_flags.append(build.DISALLOW_UNTYPED_CALLS)
+
+    if args.disallow_untyped_defs:
+        options.build_flags.append(build.DISALLOW_UNTYPED_DEFS)
+
+    # experimental
+    if args.fast_parse:
+        options.build_flags.append(build.FAST_PARSER)
+
+    # Set reports.
+    for flag, val in vars(args).items():
```

Comment:
I think it would be a bit unfortunate if we had to list out all the reports twice (but I'm not strongly opposed).  I only used `vars()` here because I was explicitly told to do this in the argparse documentation! https://docs.python.org/3/library/argparse.html#argparse.Namespace